pkg/dashboard/resource/apigateway.go

```diff
@@ -366,13 +371,18 @@ func (agr *apiGatewayResource) processAPIGatewayInfo(apiGatewayInfoInstance *api
 	// ensure spec exists if it's required
 	if apiGatewayInfoInstance.Spec == nil {
 		if specRequired {
-			err = errors.New("Api gateway spec must be provided")
-			return nuclio.WrapErrBadRequest(err)
+			return nuclio.NewErrBadRequest("Api gateway spec must be provided")
 		}
 
 		apiGatewayInfoInstance.Spec = &platform.APIGatewaySpec{}
 	}
 
+	if specRequired {
```

Comment:
I find it a bit hard to keep track when you enrich and when you validate input, I would suggest to separate it to two function and enrich and then validate